### ISSUE TITLE
s3 backend configuration not as DRY as can be expected 

### ISSUE THREAD

**Issue by lestephane** [gruntwork-io/terragrunt#389]
(I'm a first time user, so this is more of a question)

I have this in each of my live repo subdirectories:

main.tf
----------
```hcl
terraform {
  backend "s3" {} # filled in by Terragrunt
}
```

terraform.tfvars
--------------------
```hcl
terragrunt = {
  terraform {
    source = "git@github.com:.../..-modules.git//...?ref=vX.Y.Z"
  }
  include {
    path = "${find_in_parent_folders()}"
  }
}
```
and the full s3 backend configuration is present once in the parent folder.

Is there a way to avoid copying and pasting the s3 backend stub in each subdirectory, and to rely on the s3 backend configuration present in the parent folder **by default**?

Cheers

**Comment by brikis98** [comment]
Currently, no. Terraform requires the `backend { ... }` stub to be directly in your code. So far, we've tried to make everything Terragrunt explicit, so we've avoided "magical" code generation and "magically" inheriting from parent folders. 

That said, I agree things could be more DRY, so if we can find a way to do that without magic that no one can figure out, I'm certainly open to ideas. One option we're considering is [terragrunt hooks](https://github.com/gruntwork-io/terragrunt/issues/377), which would allow you to (explicitly) generate a `backend { ... }` stub for your modules if you wanted to.
